rskj-core/src/test/java/co/rsk/peg/BridgeSupportRegisterBtcTransactionTest.java

```diff
@@ -0,0 +1,2553 @@
+
+        BridgeSupport bridgeSupport = buildBridgeSupport(activations);
+        Keccak256 flyoverDerivationHash = bridgeSupport.getFlyoverDerivationHash(
+            derivationArgumentsHash,
+            userRefundBtcAddress,
+            lpBtcAddress,
+            lbcAddress
+        );
+
+        Address flyoverFederationAddress = PegTestUtils.getFlyoverAddressFromRedeemScript(
+            bridgeMainnetConstants,
+            activeFederation.getRedeemScript(),
+            Sha256Hash.wrap(flyoverDerivationHash.getBytes())
+        );
+
+        BtcTransaction btcTransaction = new BtcTransaction(bridgeMainnetConstants.getBtcParams());
+
+        BtcECKey senderBtcKey = new BtcECKey();
+        ECKey senderRskKey = ECKey.fromPublicOnly(senderBtcKey.getPubKey());
+        RskAddress rskAddress = new RskAddress(senderRskKey.getAddress());
+        btcTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX, ScriptBuilder.createInputScript(null, senderBtcKey));
+
+        Coin minimumPeginTxValue = bridgeMainnetConstants.getMinimumPeginTxValue(activations);
+        Coin belowMinimumPeginTxValue = minimumPeginTxValue.minus(Coin.SATOSHI);
+
+        Coin amountToSend = shouldSendAmountBelowMinimum ? belowMinimumPeginTxValue : minimumPeginTxValue;
+        btcTransaction.addOutput(amountToSend, flyoverFederationAddress);
+
+        PartialMerkleTree pmt = createPmtAndMockBlockStore(btcTransaction, height);
+
+        // act
+        bridgeSupport.registerBtcTransaction(
+            rskTx,
+            btcTransaction.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize()
+        );
+
+        // assert
+        // fingerroot - unknown tx should be processed and try to register
+        if (activations == fingerrootActivations) {
+            verify(provider, times(1)).setHeightBtcTxhashAlreadyProcessed(btcTransaction.getHash(false), rskExecutionBlock.getNumber());
+            verify(bridgeEventLogger, never()).logRejectedPegin(any(), any());
+            verify(bridgeEventLogger, times(1)).logPeginBtc(rskAddress, btcTransaction, Coin.ZERO, 0);
+        }
+
+        // tbd600Activations but before grace period - unknown tx should be rejected
+        else if (activations == tbd600Activations && !shouldUsePegoutTxIndex) {
+            verify(bridgeEventLogger, never()).logPeginBtc(any(), any(), any(), anyInt());
+            verify(provider, never()).setHeightBtcTxhashAlreadyProcessed(any(), anyLong());
+            verify(bridgeEventLogger, times(1)).logRejectedPegin(btcTransaction, INVALID_AMOUNT);
+        }
+
+        // tbd600Activations and after grace period - unknown tx are just ignored
+        else {
+            verify(bridgeEventLogger, never()).logPeginBtc(any(), any(), any(), anyInt());
+            verify(bridgeEventLogger, never()).logRejectedPegin(any(), any());
+            verify(bridgeEventLogger, never()).logUnrefundablePegin(any(), any());
```

Comment:
Should we do a `verify(provider, never()).setHeightBtcTxhashAlreadyProcessed(any(), anyLong());` here too or no need?